server/src/controllers/GrailController.ts

```diff
@@ -86,8 +102,157 @@ export class GrailController {
       dataToSet.runewordData = runewordGrailData;
       return dataToSet;
     });
+
+    this.getMemberParties(address, parties => {
+      parties.forEach(party => {
+        this.saveGrailInPartyDatabase(address, party.address, grailData);
+      });
+    });
```

Comment:
honestly I don't like this code at all. This way we save a lot of redundant data.
Imho we should rather do it like this:
`Parties` collection contains this object: (basically I remove data)
``` 
export interface IPartyData {
  address: string;
  token: string;
  userlist?: any;
  pendingUserlist?: any;
  settings?: any;
}
```

On `grail`-save we then store the `data` whic hyou had in the `IPartyData` (score, missing items, etc.) in the normal use object in the grail collection, so it should look like this:

```
export interface IHolyGrail {
  address: string;
  password: string;
  token: string;
  version: string;
  data?: any;
  ethData?: any;
  runewordData?: any;
  settings?: any;
  partyData?: any
}

```
Then you only need to do 1 single db call all the time (which we already do to save the grail).

To get all the party-data for your leader board page, we then have to do 2 queries:

1. Get Party-Data from the party collection
2. Do an `in` query with mongo which contains all address from the `userList` which was returned from the first query

Then merge this data and return it to the client. Should still be fast!

Reply:
This sounds like a good solution, I'll implement for next commit.